File: app/src/main/java/niedermeyer/nonogram/gui/puzzle/GameFieldDisplayer.java
package niedermeyer.nonogram.gui.puzzle;

import android.content.Context;
import android.view.View;
import android.widget.TableLayout;

/**
 * @author Elen Niedermeyer, last modified 2022-02-15
 */
public class GameFieldDisplayer {

    private final Context context;

    public GameFieldDisplayer(Context pContext) {
        context = pContext;
    }

    public TableLayout getPuzzleView(int[][] pUserField, int pCellSize, View.OnClickListener pOnFieldClick) {
        TableLayout table = new TableLayout(context);

        // add rows of nonogram
        for (int rowIndex = 0; rowIndex < pUserField.length; rowIndex++) {
            // new row
            GameFieldRow rowView = new GameFieldRow(context);

            // add a view for each cell in this row
            for (int columnIndex = 0; columnIndex < pUserField[rowIndex].length; columnIndex++) {
                // set id such that row_index = id/100 and column_index = id%100
                GameFieldCell newCell = new GameFieldCell(context, (rowIndex * 100) + columnIndex, pCellSize, pUserField[rowIndex][columnIndex], pOnFieldClick);
                rowView.addView(newCell);
            }

            table.addView(rowView);
        }

        return table;
    }

}

File: app/src/main/java/niedermeyer/nonogram/gui/puzzle/GameFieldDisplayer.kt
package niedermeyer.nonogram.gui.puzzle

import android.content.Context
import android.view.Gravity
import android.view.View
import android.widget.LinearLayout
import niedermeyer.nonogram.logics.GameFieldCell

class GameFieldDisplayer(private val context: Context) {

    fun getPuzzleView(
        cells: List<GameFieldCell>,
        rowNumber: Int,
        cellSize: Int,
        onCellClick: View.OnClickListener
    ): LinearLayout {
        val gameFieldView = LinearLayout(context)
        gameFieldView.orientation = LinearLayout.VERTICAL
        gameFieldView.gravity = Gravity.CENTER

        // new row
        var rowView = GameFieldRowView(context)
        for (i in cells.indices) {
            val newCell = GameFieldCellView(
                context,
                i,
                cellSize,
                cells[i].userValue,
                onCellClick
            )
            rowView.addView(newCell)

            if ((i + 1) % rowNumber == 0) {
                gameFieldView.addView(rowView)
                rowView = GameFieldRowView(context)
            }
        }

        return gameFieldView
    }

}